### PR TITLE
rpcserver: Make sure the arguments are provided when calling `DeleteAllPayments` RPC

### LINKED ISSUE #5641: Suggestion: REST API - Deleting payment arguments should not be optional.
### Background

Since the Delete Payments operation is so destructive, the arguments should not be optional. If not defined, I think it would be better to return an error.

REST API - [DELETE] https://api.lightning.community/#v1-payments

### Your environment

LND version 0.13.1-beta

### More Info

The documentation does not define these arguments as optional, nor document the defaults either

### PR DESCRIPTION
**Pull request by mohamedawnallah**
## Change Description

Closes #5641

## Steps to Test
Steps for reviewers to follow to test the change.

### Testing
- [ ] Your PR passes all CI checks.
- [ ] Tests covering the positive and negative (error paths) are included.
- [ ] Bug fixes contain tests triggering the bug to prevent regressions.

### Code Style and Documentation
- [ ] The change obeys the [Code Documentation and Commenting](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#CodeDocumentation) guidelines, and lines wrap at 80.
- [ ] Commits follow the [Ideal Git Commit Structure](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#IdealGitCommitStructure).
- [ ] Any new logging statements use an appropriate subsystem and logging level.
- [ ] Any new lncli commands have appropriate tags in the comments for the rpc in the proto file.
- [ ]  [There is a change description in the release notes](https://github.com/lightningnetwork/lnd/tree/master/docs/release-notes), or `[skip ci]` in the commit message for small changes.

📝 Please see our  [Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md) for further guidance.

### CODE REVIEW

File: lnrpc/lightning.proto [hunk 1]
```diff
@@ -4231,6 +4231,10 @@ message DeleteAllPaymentsRequest {
     Only delete failed HTLCs from payments, not the payment itself.
     */
     bool failed_htlcs_only = 2;
+
+    // Delete all payments. NOTE: Using this option requires careful
+    // consideration as it is a destructive operation.
+    bool all_payments = 3;
```
Comment (guggero) [inline lnrpc/lightning.proto:4237]: Doesn't this also require a change in the `cmd/lncli/cmd_payments.go` file?

**Review by guggero — COMMENTED** [review]
Thanks for the PR! Definitely a good safety feature. But I think we also need an update to `lncli`?

File: lnrpc/lightning.proto (hunk 1 above)
Comment (mohamedawnallah) [reply lnrpc/lightning.proto:4237]: Thanks! Missed that. Updated!

**Review by guggero — APPROVED** [review]
Very nice! tACK, LGTM :tada:

File: rpcserver.go [hunk 2]
```diff
@@ -6757,6 +6757,27 @@ func (r *rpcServer) DeleteAllPayments(ctx context.Context,
...
+	// Since this is a destructive operation, at least one of the options
+	// must be set to true.
+	case !req.AllPayments && !req.FailedPaymentsOnly &&
+		!req.FailedHtlcsOnly:
+
+		return nil, fmt.Errorf("at least one of the options " +
+			"`all_payments`, `failed_payments_only`, or " +
+			"`failed_htlcs_only` must be set to true")
+
+	// 'all_payments' cannot be true with 'failed_payments_only' or
```
[the remark this answers is not in the document]
Comment (mohamedawnallah) [reply rpcserver.go]: Thanks! Updated.

File: lnrpc/lightning.proto [hunk 3]
```diff
@@ -4231,6 +4231,10 @@ message DeleteAllPaymentsRequest {
     Only delete failed HTLCs from payments, not the payment itself.
     */
     bool failed_htlcs_only = 2;
+
```
Comment (yyforyongyu) [inline lnrpc/lightning.proto:4234]: maybe the `DeleteAllPayments` should be renamed to `DeletePayments` for clarity - future PRs tho.

**Comment by mohamedawnallah** [comment]
Thanks @yyforyongyu! Rebased.

cc @guggero

File: lnrpc/lightning.proto (hunk 3 above)
Comment (guggero) [reply lnrpc/lightning.proto:4234]: Unfortunately we can't really do that without breaking older RPC clients... But yeah, would've been more precise.

**Review by guggero — APPROVED** [review]

File: lnrpc/lightning.proto (hunk 3 above)
Comment (mohamedawnallah) [reply lnrpc/lightning.proto:4234]: I thought of the backward compatibility as well 👍

**Pull request merged**